File: output/ChatGPT/codenet/Python/C++/s289483518.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n, m;
    cin >> n >> m;
    vector<int> a(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }
    sort(a.begin(), a.end());
    vector<vector<int>> ope(m, vector<int>(2));
    for (int i = 0; i < m; i++) {
        cin >> ope[i][0] >> ope[i][1];
    }
    sort(ope.begin(), ope.end(), [](const vector<int>& x, const vector<int>& y) {
        return x[1] > y[1];
    });
    int j = 0;
    for (int i = 0; i < n; i++) {
        if (a[i] < ope[j][1]) {
            a[i] = ope[j][1];
            ope[j][0]--;
            if (ope[j][0] == 0) {
                j++;
                if (j == ope.size()) {
                    break;
                }
            }
        }
        else {
            break;
        }
    }
    int sum = 0;
    for (int i = 0; i < n; i++) {
        sum += a[i];
    }
    cout << sum << endl;

    return 0;
}

File: output/ChatGPT/codenet/Python/Java/s289483518.java
import java.util.Arrays;
import java.util.Comparator;
import java.util.Scanner;

public class s289483518 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int m = scanner.nextInt();
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = scanner.nextInt();
        }
        Arrays.sort(a);
        int[][] ope = new int[m][2];
        for (int i = 0; i < m; i++) {
            ope[i][0] = scanner.nextInt();
            ope[i][1] = scanner.nextInt();
        }
        Arrays.sort(ope, Comparator.comparingInt(o -> o[1]));
        int j = 0;
        for (int i = 0; i < n; i++) {
            if (a[i] < ope[j][1]) {
                a[i] = ope[j][1];
                ope[j][0]--;
                if (ope[j][0] == 0) {
                    j++;
                    if (j == ope.length) {
                        break;
                    }
                }
            } else {
                break;
            }
        }
        int sum = 0;
        for (int i = 0; i < n; i++) {
            sum += a[i];
        }
        System.out.println(sum);
    }
}